### ISSUE TITLE
`RANSAC.forward` returns an (N, 1) inlier mask when no model is found and (N,) otherwise

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4871]
`RANSAC.forward` initialises its result mask as `torch.zeros(num_tc, 1, dtype=torch.bool)`. It replaces that mask with the `(N,)` mask from `verify` only when a model is accepted. The mask's shape therefore depends on the branch taken: the usual `kp1[mask]` works when a model is found and raises `IndexError` when none is, the case a caller most needs to handle. The docstring matches neither branch: it promises a model of shape `(1, 3, 3)` and a mask of shape `(1, N)`, while the model is `(3, 3)` in both cases. A function's output shape should not depend on whether it succeeded.

## Repro

```python
import torch
from kornia.geometry.linalg import transform_points
from kornia.geometry.ransac import RANSAC

H = torch.tensor([[[1.034685, -0.04160598, -63.61482],
                   [0.1039753, 1.135451, -62.87412],
                   [2.851519e-04, 1.404178e-04, 1.0]]])
gx, gy = torch.tensor([160.0, 230.0, 330.0, 440.0]), torch.tensor([95.0, 150.0, 240.0, 320.0])
p1 = torch.stack(torch.meshgrid(gx, gy, indexing="xy"), -1).reshape(16, 2) + torch.linspace(0, 3, 16)[:, None]
p2 = transform_points(H, p1[None])[0]
g = torch.Generator().manual_seed(3)
r1, r2 = torch.rand(16, 2, generator=g) * 500, torch.rand(16, 2, generator=g) * 500   # no consistent model

ransac = RANSAC("homography", inl_th=0.5, seed=0)
for name, (a, b) in [("model found", (p1, p2)), ("no model found", (r1, r2))]:
    model, mask = ransac(a, b)
    print(f"{name:15}: model {tuple(model.shape)} all-zero={bool((model == 0).all())!s:5} | mask {tuple(mask.shape)} {mask.dtype}")
    try:
        print("   p1[mask].shape =", tuple(p1[mask].shape))
    except Exception as e:
        print("   p1[mask] ->", type(e).__name__, str(e).splitlines()[0])
```

```text
model found    : model (3, 3) all-zero=False | mask (16,) torch.bool
   p1[mask].shape = (16, 2)
no model found : model (3, 3) all-zero=True  | mask (16, 1) torch.bool
   p1[mask] -> IndexError The shape of the mask [16, 1] at index 1 does not match the shape of the indexed tensor [16, 2] at index 1
```

kornia `main` 7ddf731bd, torch 2.14.0 (and 2.5.1: same output), CPU float32, macOS arm64.

Inside kornia, `ImageStitcher` discards the mask and `PlanarTracker` only calls `.sum()` on it, so neither is affected. The `(N, 1)` initialiser dates from #1335 (2021).

## Classification

API contract gap: the return shape differs between the success and failure paths, and the docstring's shapes match neither.

No idiom of 1–9; candidate: output shape depends on the branch taken.

## Disposition

Focused fix, welcome as a PR: initialise the mask as `torch.zeros(num_tc, dtype=torch.bool, device=kp1.device)`, and correct the `forward` docstring to a model of shape `(3, 3)` (all zeros when no model is found) and a mask of shape `(N,)`.

Found by the batch-8 conventions audit (two-view geometry); a `test_wart_*` pin in the docs PR flips when the fix lands.

Related: #4868 (MSAC failures return this mask); #4872 (another RANSAC input/output contract gap).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
